concepts/infrastructure-reconciliation/README.md

```diff
@@ -0,0 +1,35 @@
+# Motivation
+Reconciler is a framework to install, update and repair Kyma components in managed Kyma Runtimes. The reconciliation runs in the loop to ensure that all components are properly configured, up and running. This principle can be extended to cover also infrastructure components like virtual networks or kubernetes clusters, but requires some adjustments in the reconciler architecture. 
+
+# New use cases for reconciler
+
+- create VNET for Kyma Runtime 
+- create/reconcile shoot cluster for Kyma Runtime 
+- produce kubeconfig for provisioned clusters (can be rotated)
+- support bring your own cluster models (BYOC) - cluster is ready and we got cluster-admin kubeconfig 
+- trigger reconciliation on configuration or cluster state change
+
+# Proposed changes
+
+## Cluster Inventory as a separate service
+
+Cluster Inventory is the internal component responsible for reconciliation persistence. It keeps cluster information, configuration and reconciliation status. Component reconcilers do not have access to the inventory and cannot modify it directly. One option would be to extend action context passed to the reconciler to include more data, but the better option would be to pass only information how to access cluster inventory entry (url/id) and allow components reconciler to query and update it.
+
+## Watches and triggers
```

Comment:
I have one question concerning watcher in Kyma cluster. I assume that the watcher will observe some limited subset of resources to trigger reconciliation immediately in some cases (e.g. certs for Istio Gateway deleted). In most cases modifications performed by the user will be restored by time based trigger. Is that correct?

Reply:
Plan is to watch all resources which have a particular label. The reconciler will frequently (e.g. each minute) check for identified changes and trigger then a reconciliation (we have to use pull-communication caused to security restrictions: access from SAP networks to the Internet is acceptable but not vs.).